File: VB/ChangeEditMask.Module/BusinessObjects/DemoObject.vb
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Persistent.Base
Imports DevExpress.Xpo

Namespace ChangeEditMask.Module.BusinessObjects

    <DefaultClassOptions>
    Public Class DemoObject
        Inherits BaseObject

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub

        Private _Mask As EditMask

        <ImmediatePostData>
        Public Property Mask As EditMask
            Get
                Return _Mask
            End Get

            Set(ByVal value As EditMask)
                SetPropertyValue("Mask", _Mask, value)
            End Set
        End Property

        Private _TestString As String

        Public Property TestString As String
            Get
                Return _TestString
            End Get

            Set(ByVal value As String)
                SetPropertyValue("TestString", _TestString, value)
            End Set
        End Property
    End Class

    Public Enum EditMask
        [Date]
        Time
        Numeric
        [String]
    End Enum
End Namespace

File: CS/XPO/DynamicMask/DynamicMask.Module/BusinessObjects/DemoObject.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Persistent.Base;
using DevExpress.Xpo;
using DevExpress.ExpressApp.Model;

namespace DynamicMask.Module.BusinessObjects {
    [DefaultClassOptions]
    public class DemoObject : BaseObject {
        public DemoObject(Session session) : base(session) { }
        private EditMask _Mask;
        [ImmediatePostData]
        public EditMask Mask {
            get { return _Mask; }
            set { SetPropertyValue("Mask", ref _Mask, value); }
        }
        private string _TestString;
        [ModelDefault("EditMask", "(000) 000-0000")]
        public string TestString {
            get { return _TestString; }
            set { SetPropertyValue("TestString", ref _TestString, value); }
        }
    }
    public enum EditMask {
        Date,
        Time,
        Numeric,
        String
    }
}
